src/features/toast/ToastRenderer.tsx

```diff
@@ -1,41 +1,85 @@
-import React, { useRef } from 'react'
+import React, { useEffect, useRef, useState } from 'react'
 import { useSelector } from 'react-redux'
 import { ToastContainer, toast } from 'react-toastify'
 
 import { Run, RunStatus } from '../../qri/run'
 import Toast from './Toast'
 import { selectRunsFromEventLog } from '../workflow/state/workflowState'
+import { API_BASE_URL } from "../../store/api"
+import { VersionInfo } from "../../qri/versionInfo"
+import { selectCollection } from "../collection/state/collectionState"
 
 
 import 'react-toastify/dist/ReactToastify.css'
 
+
 const ToastRenderer: React.FC<{}> = () => {
 
   const runs = useSelector(selectRunsFromEventLog)
-
+  const initialCollections = useSelector(selectCollection)
   // use a ref to store the workflow status so we can
   // determine when it changes and update an already visible toast
   const runsRef = useRef({} as Record<string, RunStatus>)
 
+  const [ collections, setCollections ] = useState<VersionInfo[]>([])
+  const [ initIds, setInitIds ] = useState<string[]>([])
+
   // artifically delay the update to 'success'
   const SUCCESS_START_DELAY = 0
 
   // after 'success', close the toast 4 seconds later
   const SUCCESS_CLOSE_DELAY = 4000
 
+  useEffect(() => {
+    initIds.forEach( async (initId) => {
+      let isResponse = collections.find(c => c.initID === initId)
+      if(!isResponse && initialCollections.length)
+        isResponse = initialCollections.find(c => c.initID === initId)
+      if(!isResponse){
+        const initialResponse = await fetch(`${API_BASE_URL}/collection/get`, {
+          method: 'POST',
+          body: JSON.stringify({initID: initId})
+        })
+        const response:VersionInfo = await initialResponse.json().then(r => r.data)
+        if(response){
+          setCollections([...collections, response])
+        }
+      }
+    })
```

Comment:
usually when we see fetching happening in the component along the use of `useState`, it usually means we are missing something in redux. I proposed a large-ish refactor here that changes the way we think of the `state.collection.collection` & uses it as a place to store version infos, rather than the place that tells us what collections to show on the "My Collections" page. It would also mean a bunch of changes to this `ToastRenderer` component: https://github.com/qri-io/frontend/issues/261#issuecomment-928447901

LMK if you have any questions!